crates/tests/result/tests/error.rs

```diff
@@ -10,29 +10,30 @@ windows_targets::link!("kernel32.dll" "system" fn SetLastError(code: u32));
 
 #[test]
 fn empty() {
-    const EMPTY: Error = Error::empty();
-    assert_eq!(EMPTY.code(), S_OK);
-    assert!(EMPTY.as_ptr().is_null());
-    assert_eq!(EMPTY.message(), "The operation completed successfully.");
+    let e: Error = Error::empty();
+    assert_eq!(e.code(), E_FAIL);
```

Comment:
This feel like a pretty non-obvious breaking change - I know that we'll be bumping the version number to indicate a breaking change, but still this should be noted somewhere...

Reply:
Not sure I agree with this change; `Error::default() / empty()` defaulting to non-zero is an unexpected behavior. If we can't preserve this, then I'd be in favor of abandoning the entire PR.

Reply:
It's documented on `Error::new` and the other paths that are affected.

I agree it's a breaking change, but a well-deserved ones.  Errors should be...  errors.

Reply:
Why is there even an `Error::empty()` at all?  What does an "empty error" even mean?  I think `Error::empty()` should be removed entirely.

Reply:
> Why is there even an `Error::empty()` at all? What does an "empty error" even mean? I think `Error::empty()` should be removed entirely.

+1 I'd prefer to remove it than to change its meaning - at least that'd cause build breaks.

Reply:
Switched to `Default`.  Renamed `Error::empty()` to `Error::fail()`.  Changed several usages of `Error::empty()` to `Error::from_hresult(E_POINTER)`, which is much more accurate.